src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,492 @@
+
+    val prisonersWithoutLicences = prisonersApproachingRelease.filter { p ->
+      !licenceNomisIds.contains(p.prisoner.prisonerNumber)
+    }
+    val eligiblePrisoners = filterOffendersEligibleForLicence(prisonersWithoutLicences.toList())
+    val casesWithoutLicences = pairNomisRecordsWithDelius(eligiblePrisoners)
+    return createNotStartedLicenceForCase(casesWithoutLicences)
+  }
+
+  private fun formatReleasedLicences(licences: List<LicenceSummary>): List<CaCase> {
+    val groupedLicences = licences.groupBy { it.nomisId }
+    return groupedLicences.map {
+      val licence = if (it.value.size > 1) {
+        it.value.find { l -> l.licenceStatus != ACTIVE }
+      } else {
+        it.value[0]
+      }
+      val releaseDate = licence?.actualReleaseDate ?: licence?.conditionalReleaseDate
+      CaCase(
+        kind = licence?.kind,
+        licenceId = licence?.licenceId,
+        licenceVersionOf = licence?.versionOf,
+        name = "${licence?.forename} ${licence?.surname}",
+        prisonerNumber = licence?.nomisId!!,
+        releaseDate = releaseDate!!,
+        releaseDateLabel =
+        when (licence.actualReleaseDate) {
+          null -> "CRD"
+          else -> "Confirmed release date"
+        },
+        licenceStatus = licence.licenceStatus,
+        lastWorkedOnBy = licence.updatedByFullName,
+        isDueForEarlyRelease = licence.isDueForEarlyRelease,
+        isInHardStopPeriod = licence.isInHardStopPeriod,
+        probationPractitioner = ProbationPractitioner(
+          staffUsername = licence.comUsername,
+        ),
+      )
+    }
+  }
+
+  fun splitCasesByComDetails(cases: List<CaCase>): GroupedByCom = cases.fold(
+    GroupedByCom(
+      withStaffCode = emptyList(),
+      withStaffUsername = emptyList(),
+      withNoComId = emptyList(),
+    ),
+  ) { acc, caCase ->
+    if (caCase.probationPractitioner?.staffCode != null) {
+      acc.withStaffCode += caCase
+    } else if (caCase.probationPractitioner?.staffUsername != null) {
+      acc.withStaffUsername += caCase
+    } else {
+      acc.withNoComId += caCase
+    }
+    return@fold acc
+  }
+
+  private fun createNotStartedLicenceForCase(cases: List<ManagedCase>): List<CaCase> = cases.map { c ->
+
+    // Default status (if not overridden below) will show the case as clickable on case lists
+    var licenceStatus = NOT_STARTED
+
+    if (isBreachOfTopUpSupervision(c)) {
+      // Imprisonment status indicates a breach of top up supervision order - not clickable (yet)
+      licenceStatus = OOS_BOTUS
+    } else if (isRecall(c)) {
+      // Offender is subject to an active recall - not clickable
+      licenceStatus = OOS_RECALL
+    } else if (c.cvlFields.isInHardStopPeriod) {
+      licenceStatus = TIMED_OUT
+    }
+
+    val com = c.deliusRecord?.managedOffenderCrn?.staff
+
+    val releaseDate = c.nomisRecord?.confirmedReleaseDate ?: c.nomisRecord?.conditionalReleaseDate
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**suggestion:** ‏Again, could use `getLicenceStartDate` against the `PrisonerSearchPrisoner` result here.